datafusion/physical-plan/src/topk/mod.rs

```diff
@@ -374,6 +368,12 @@ impl TopK {
             return Ok(());
         }
 
+        // Extract scalar values BEFORE acquiring lock to reduce critical section
+        let thresholds = match self.heap.get_threshold_values(&self.expr)? {
+            Some(t) => t,
+            None => return Ok(()),
```

Comment:
this was an early return, but I was thinking that extracting the values can be delayed until after we decided if an update is needed